### PR TITLE
SLE 2024-04 release notes

### PR DESCRIPTION
**Pull request by barnabas-nagy-ni**
Why should this Pull Request be merged?
Contains release notes for 2024-04 SLE release.

What testing has been done?
NA

### CODE REVIEW

File: release-notes/2024-04/README.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# SystemLink Enterprise release 2024-04 Release Notes
+
+The 2024-04 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.
+
+## New Features and Behavior changes
```
Comment (prestwick) [inline release-notes/2024-04/README.md:5]: Please match the text for new features as documented in https://dev.azure.com/ni/Users/_git/Documentation%20Reviews?path=/review%20output/systemlink-enterprise%20md/SystemLink%20Enterprise%20New%20Features%20and%20Changes/systemlink-enterprise-new-features-and-changes.md

File: release-notes/2024-04/README.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        - feed:Delete
+- `feedsui 0.0.364`
+    - New helm chart added for feeds.
+- `serviceregistry 0.13.12`
+    - Added new `testplan:Execute` permission to allow calling the new `/niworkorder/v1/testplans/{testPlanId}/execute` route. The Work Order service is not installed by default. This route is marked "[Beta]" and is still in development.
+
+## Helm Chart Breaking Changes
+
+- `workorder 0.3.79`
+    - workorder database needs to be dropped on upgrade.
```
Comment (prestwick) [inline release-notes/2024-04/README.md:21]: How does a customer do this. What is our recommended procedure for doing so. If this process is long we should ask the team responsible for work-order to draft a doc on how to do this and reference it from here.

File: release-notes/2024-04/README.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+- `workorder 0.3.79`
+    - Status property has been renamed to state in workorder and testplan APIs. In the future, Status will have a different meaning.
```
Comment (prestwick) [inline release-notes/2024-04/README.md:23]:
```suggestion
    - The `status` property has been renamed to `state` in `workorder` and `testplan` APIs.
```

**Review by prestwick — CHANGES_REQUESTED** [review]

File: release-notes/2024-04/README.md (hunk 2 above)
Comment (chris468) [reply release-notes/2024-04/README.md:29]: Basically they stop the work order deployment, log in to the mongo pod, and use mongosh commands to drop the db. I will push a document w/ instructions and a link to this branch later today.

Comment (chris468) [reply release-notes/2024-04/README.md]: I pushed a draft and added a link.

File: release-notes/2024-04/Drop-workorder-database.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Dropping the work order database
+
+## Assumptions
+
+- The workorder service is enabled but not in use, no data needs to be retained.
+- SystemLink Enterprise deployment uses in-cluster, per-service mongo databases.
+- Administrator has `kubectl` access to the Kubernetes cluster from a local bash shell.
```
Comment (chris468) [inline release-notes/2024-04/Drop-workorder-database.md:7]: @prestwick Is assuming local bash OK? Quoting would be different if it's cmd or pwsh.

File: release-notes/2024-04/Drop-workorder-database.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Dropping the work order database
+
+## Assumptions
+
+- The workorder service is enabled but not in use, no data needs to be retained.
+- SystemLink Enterprise deployment uses in-cluster, per-service mongo databases.
```
Comment (chris468) [inline release-notes/2024-04/Drop-workorder-database.md:6]: @prestwick Do we need instructions for any other cases besides per-service db instance?

File: release-notes/2024-04/Drop-workorder-database.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
e Kubernetes cluster from a local bash shell.
+- Administrator has permission to stop and start deployments and to exec into pods.
+
+## Procedure
+
+1. Complete the upgrade to the 2024-04 helm chart
+1. Stop the work order deployment
+   `kubectl scale deployment <release-name>-workorder` --replicas=0
+1. Wait for all `<release-name>-workorder` pods to stop.
+1. Determine which mongo pod is the primary replicaset.
```
Comment (prestwick) [inline release-notes/2024-04/Drop-workorder-database.md:16]:
```suggestion
1. Determine which MongoDB pod is the primary replicaset.
```

Also, is each replica its own pod or are they collectively a pod?

**Review by prestwick — CHANGES_REQUESTED** [review]

File: release-notes/2024-04/Drop-workorder-database.md (hunk 6 above)
Comment (chris468) [reply release-notes/2024-04/Drop-workorder-database.md:16]: I meant "replica" here, not "replicaset". Each replica is a pod.

File: release-notes/2024-04/Drop-workorder-database.md (hunk 5 above)
Comment (chris468) [reply release-notes/2024-04/Drop-workorder-database.md]: Added external

**Review by prestwick — APPROVED** [review]

File: release-notes/2024-04/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
+# SystemLink Enterprise release 2024-04 Release Notes
+
+The 2024-04 release bundle for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.
+
+## New Features and Behavior changes
+
+- [Create test plans](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/creating-a-test-plan.html) faster by using a test plan template.
```
[the remark this answers is not in the document]
Comment (prestwick) [reply release-notes/2024-04/README.md:7]: Discussed with Josh to use this description in a preamble for the docs we have on test plans. That will happen after the release notes are finalized.

**Pull request merged**